### ISSUE TITLE
Why is TRAINING RFDETRSegPreview on large images terrible slow?

### ISSUE THREAD

**Issue by Jonas17012002** [roboflow/rf-detr#493]
Training on 1024x1024 images takes  0.1 seconds per image.  On 2448x2048 it takes 1 second per image
Considering, that training happens on a  504x504 resolution in BOTH cases this seems to be odd. 

the train script: 

```

import torch
import time
import gc
import os
import sys
sys.path.append("rf-detr-repo")
from rfdetr import RFDETRSegPreview

dataset_dir = "custom/path"
output_dir = "custom/path"

gc.collect()
torch.cuda.empty_cache()

model = RFDETRSegPreview()  # args in kwargs umwandeln


gc.collect()
torch.cuda.empty_cache()

model.train(
    device="cuda:0",
    dataset_dir=dataset_dir,
    resolution=504,  #Must be divisible by 24
    batch_size=1,
    grad_accum_steps=4,
    epochs=5,     
    lr=1e-4,
    lr_backbone=1e-5,
    output_dir=output_dir,
    #resume_dir=output_dir,     
    freeze_backbone=True, 
    fp16=True,
    square_resize_div_64 = True,   #For some weird reason that is set to True, per default. 
    multi_scale=False,
    do_random_resize_via_padding=False,
)
```

First Idea might be the preprocessing, but only basic transformations are used.
When I only use normalize, execution speed doesn't change.
This is the transformation code from  datasets/coco.py  in make_coco_transforms_square_div_64:

```
 if image_set == 'train':
        return T.Compose([
            T.RandomHorizontalFlip(),
            T.RandomSelect(
                T.SquareResize(scales),
                T.Compose([
                    T.RandomResize([400, 500, 600]),
                    T.RandomSizeCrop(384, 600),
                    T.SquareResize(scales),
                ]),
            ),
            normalize,
        ])
        ])
```

I collected the time information by adding a print to the __getitem__ method from the CocoDetection class in datasets/coco.py.  All the transformations are correct (checked by visualizing as well).

```
    def __getitem__(self, idx):
        img, target = super(CocoDetection, self).__getitem__(idx)
        image_id = self.ids[idx]
        target = {'image_id': image_id, 'annotations': target}
        img, target = self.prepare(img, target)
        if self._transforms is not None:
            img, target = self._transforms(img, target)
            print("after transformation:", target['boxes'].shape, target['masks'].shape, target['masks'][0,0,0], target['masks'].sum()
        return img, target
```

Maybe it has to do something with the working principles of the DataLoader()?

Dataset information: 
- very large and very small instances
- 1 to 250 instances per image. 

I am using: 
- Linux Ubuntu x86_64
- NVIDIA RTX 5090
- Driver Version: 580.82.09    
- CUDA Version: 13.0
- AMD Ryzen 9 7950X 16-Core Processor

My enviroment: 
- conda create ... python=3.11
- pip install torch==2.6.0 torchvision pillow triton
- pip install supervision pydantic peft pycocotools
- And RFDETR mit sys.path.append("rf-detr-repo") in the working directory
